lib/AI-Agent.xml

```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ApplicationInsightsAgent>
+    <AgentLogger>INFO</AgentLogger>
+    <Instrumentation>
+        <BuiltIn enabled="true">
+            <HTTP enabled="true"/>
+        </BuiltIn>
+        <Class name="uk.gov.hmcts.ccd.data.casedetails.CaseAuditEventRepository">
+            <Method name="set"/>
+            <Method name="findByCase"/>
+            <Method name="getCreateEvent"/>
+            <Method name="findByEventId"/>
+        </Class>
+        <Class name="uk.gov.hmcts.ccd.data.definition.DefaultCaseDefinitionRepository">
+            <Method name="getCaseTypesForJurisdiction"/>
+            <Method name="getCaseType"/>
+            <Method name="getBaseTypes"/>
+            <Method name="getUserRoleClassifications"/>
+            <Method name="getClassificationsForUserRoleList"/>
+            <Method name="getLatestVersion"/>
+            <Method name="getJurisdictions"/>
+        </Class>
+        <Class name="uk.gov.hmcts.ccd.data.casedetails.DefaultCaseDetailsRepository">
+            <Method name="set"/>
+            <Method name="findById"/>
+            <Method name="findByReference"/>
```

Comment:
@timja we have overloaded findByReference with different number of parameters. Will this line instrument them all?

Reply:
I think so, you can pick and choose with something like this:
https://github.com/hmcts/cmc-claim-store/blob/master/lib/AI-Agent.xml#L30

useful if i.e. all of the overloads delete to one method and then you just put the method they all call there